README.md

```diff
@@ -1,27 +1,38 @@
-# epm
-
-Ethereum Package Managment
+# EPM
 
 [![Join the chat at https://gitter.im/ethpm/Lobby](https://badges.gitter.im/ethpm/Lobby.svg)](https://gitter.im/ethpm/Lobby?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
 
-# Goals
-
-The goal of this repository is to create a specification that can be used for
-package management for the Ethereum Virtual Machine (EVM).
+Ethereum Packaging Specification
 
-This explicitely includes:
 
-* Specification for a *manifest* file.
+## Use Cases
 
+The following use cases were used to guide the specification.
 
-This explicitely does not include:
+1. Code with zero deployed instances.
+    1. Reusable base contracts.
+2. Code with one or more addresses.
+    1. Libraries: Installation of the package involves having an importable interface for the library and linking against the proper address during deployment.
+    2. Contracts: Installation of the package involves having an importable interface and some manner of templating in the proper addresses into the local source code prior to compilation as well as other non-contract code that may interact with the installed contract.
 
-* ????
 
+It is worth pointing out that the *Library* and *Contract* use cases are functionally the same.  Once [Solidity issue #242](https://github.com/ethereum/solidity/issues/242) has been addressed this distinction may no longer need to exist.
 
-# Resources
 
-The following resources are encouraged reading before joining the discussion.
+## Open Questions
 
-* https://github.com/purescript/purescript/issues/631
-    * Long discussion on package management from the purescript community.
+* Version
+    * Should the version live in the package manifest, release manifest, or both.
+        * Case for Package
+            * Defines the *current* version of the package.
+        * Case for Release
+            * Each release should have a clearly defined version.
+        * Case for Both
+            * Seems to work well for release workflow. Package manager would
+              populate the release manifest version field from the package
+              manifest version when creating a new release.
+* Validation of the concept of Package Dependencies vs Release Dependencies
+    * When a dependency is defined at the package level it *can* be either pinned to a specific release, or possibly a range of releases such as `^0.4.0` to allow any version in the `0.4.x` line.
```

Comment:
>I understand that allowing non-semver version numbering adds complexity to package manager implementations but I'm still not comfortable requiring semver since I know of plenty of major software packages that use alternate versioning schemes.

I actually don't care if it's semver or not. The issue is your proposal allows for two different packages to resolve to the same identifier for no practical reason.

>I understand what you are talking about with your FreeBSD example. I just don't understand at all how it applies to this specification.

Again, just trying to get *global* package identifiers

>The current specification allows for un-ambiguously pointing at a specific release via an IPFS URI.

I think a blockchain URI makes more sense than an IPFS one. I think relying on IPFS in the spec is kinda lazy, We should make the content-addressing hash pluggable.

>Would it your stance on this if the spec was modified such that in the cases that a dependency was declared as a package_name/version pair that we supported some sort of content-hash as well so that once the dependency has beel resolved a package manager can then verify that the resolved package is the expected package?

I think a BIP122 URI is better than a protocol-dependent content-hash but a protocol independent content-hash is at least as good as a BIP122 URI if not significantly better. I understand Swarm has adopted IPFS' content-hashing so if the spec is:

`<package_name>==<semver>#<pluggable_content_hash>`

this makes sense to me. But mixing hashes and URIs seems sort of silly. either map the global URIs to the content-hashes or just use content-hashes. I think we should just refer to all external objects via URIs. or things that are trivial to convert to URI, content-hashes have this property, but they aren't human readable, I think putting the mapping from human readable to content-hash in this file is kinda weird... but I'd need to think about it more to be sure.